URIs.md

```diff
@@ -0,0 +1,103 @@
+
+*Non-Requirements*
+
+- *Canonical Representation*: The protocol need not require all parties produce identical identifiers for the same resource. Canonicalization is necessary where different parties must independently produce identical representations of the same object (as when used for hash verification), and assists equality checking by allowing comparison of canonical identifiers. Certain resources, especially within the web, cannot be coerced into using a single canonical identifier. Nonetheless, the standard should minimize the space of valid identifiers for a single resource and should impose unique identifiers where possible.
+
+## Specification
+
+The standard is an extension of the common [URL standard](https://datatracker.ietf.org/doc/html/rfc1738) with support for the HTTP, HTTPS, and [IPFS](https://docs.ipfs.io/how-to/address-ipfs-on-web/#native-urls) protocols as well as the newly introduced `chain` and `farcaster` schemes.
+
+```
+<protocol>://<resource>,  <= 2048 characters
+
+protocol      =  "https" | "http" | "ipfs" | "chain" | "farcaster"
+resource      = (defined by each protocol specification)
+```
+
+The Farcaster URL structure adds the following additional protocols to the URL prefix:
+
+- `chain://`
+- `farcaster://`
+
+### Chain Resources
+
+Resources and activity across different networks will be identified by a URI with the `chain:` protocol, followed by the network's [CAIP-2](https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-2.md) chain ID.
+
+Following the chain ID may be a [CAIP-10](https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-10.md) account specifier, a [CAIP-19](https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-19.md) asset specifier, a transaction identifier, or other unspecified identifiers for on-chain resources and activity.
+
+```
+chain://<network>
+chain://<network>:<address>
+chain://<network>/<asset_namespace>:<asset_reference>/...
+chain://<network>/tx:<tx_identifier>
+chain://<network>/<future_placeholder>
+
+network             = CAIP-2 chain id
+address             = an address string in the common format of the given network
+asset_namespace     = an identifier for a class of related assets, such as erc20 and slip44
+asset_reference     = identifies an asset within its asset_namespace
+tx_identifier       = a transaction identifier in the common format of the given network
+future_placeholder  = future message types
+...                 = addition uri parameters for the standard (optional)
+```
+
+Examples:
+
+```
+chain://eip155:1:0xaba7161a7fb69c88e16ed9f455ce62b791ee4d03
+chain://eip155:1/erc721:0xaba7161a7fb69c88e16ed9f455ce62b791ee4d03/7894
+chain://bip122:000000000019d6689c085ae165831e93/slip44:0
+```
+
+### Farcaster Resources
+
+Farcaster URIs must be capable of identifying any message signed by a user. A sample URI might look like `farcaster://user:alice/cast:0xf00b4r/42` and is of the form:
```

Comment:
1. with the new identity model, usernames are no longer durable and can change at any time. the right identifier to use here would be the new `id` identifier that every farcaster user must have.  e.g. `farcaster://id:8789213729/cast:0xf00b4r/42`


2. With the removal of signed chains, there are no longer sequence id's and so we can deprecate the <sequence> parameter. 

3. I wonder if we need to link to the sub-type (e.g. cast-short) instead of just the main type (cast) for efficiently. See [this doc](https://www.notion.so/farcasterxyz/Spec-Farcaster-Message-c71fa4c4334542e9a3dc678be4df6fe2#8f1e622d14e347cbb062958faec24b23) for list of current sub-types. There are two benefits to this -- (1) we can prevent linking to invalid sub-types like a delete and (2) it might be faster for nodes to retrieve data from a subtype if the data is sharded across sub-types. The main  argument against it is that it creates  longer URL's and there is no way for a node to verify this on ingestion since it may not know the target that is being references.

Reply:
> (1) we can prevent linking to invalid sub-types like a delete
A good point. Also true that checking for links to invalid sub-types isn't possible if the linked cast hasn't already been ingested. On this decision I'm not seeing either option as obviously better than the other.

> (2) it might be faster for nodes to retrieve data from a subtype if the data is sharded across sub-types.
True, however I suspect that any implementations of sharding would divide up casts by users rather than by cast sub-types